src/nnpack/nnlib.jl

```diff
@@ -0,0 +1,86 @@
+const AA{N} = AbstractArray{Float32,N}
+const AA1 = Union{AA{2}, AA{3}, AA{4}, AA{5}}
+
+#NOTE: Commenting out the activation functions until sure what to do
+
+# function relu(x::AA1; inplace::Bool = true, nthreads = NNPACK_CPU_THREADS)
+#     threadpool = nthreads != NNPACK_CPU_THREADS ? pthreadpool_create(nthreads) : shared_threadpool
+#     nnp_relu_output(x, inplace ? x : similar(x), threadpool = threadpool)
+# end
+#
+# function leakyrelu(x::AA1, a = oftype(x/1, 0.01); inplace::Bool = true, nthreads = NNPACK_CPU_THREADS)
+#     threadpool = nthreads != NNPACK_CPU_THREADS ? pthreadpool_create(nthreads) : shared_threadpool
+#     nnp_relu_output(x, inplace ? x : similar(x), negative_slope = a, threadpool = threadpool)
+# end
+
+function softmax!(x::AbstractVecOrMat{Float32}; inplace::Bool = true, nthreads = NNPACK_CPU_THREADS)
```

Comment:
The `inplace` argument should be replaced with `softmax` and `softmax!`, like the original API